crates/re_data_ui/src/component.rs

```diff
@@ -44,6 +45,20 @@ impl DataUi for EntityComponentWithInstances {
             UiVerbosity::Reduced | UiVerbosity::All => false,
         };
 
+        // in some cases, we don't want to display all instances
+        let max_row = match verbosity {
+            UiVerbosity::Small => 0,
+            UiVerbosity::Reduced => num_instances.at_most(4), // includes "…x more" if any
+            UiVerbosity::All => num_instances,
+        };
+
+        let displayed_row = if num_instances <= max_row {
+            num_instances
+        } else {
+            // this accounts for the "…x more" using a row
+            max_row - 1
+        };
```

Comment:
I think we can simplify this a lot:

```suggestion
        let num_rows_to_show = match verbosity {
            UiVerbosity::Small => 0,
            UiVerbosity::Reduced => num_instances.at_most(3),
            UiVerbosity::All => num_instances,
        };
```

Reply:
My behaviour (4 rows is strictly respected):

| 3 | 4 | 5 | 6 |
|--------|--------|--------|--------|
| x | x | x | x |
| x | x | x | x |
| x | x | x | x |
| | x | ...+2 | ...+3 |

Your behaviour (4  rows is *not* respected and the "...1 more" feels weird as you might as well put the corresponding item there):
| 3 | 4 | 5 | 6 |
|--------|--------|--------|--------|
| x | x | x | x |
| x | x | x | x |
| x | x | x | x |
| | x | x | x |
| |  | ...+1 | ...+2 |

The key difference is the `max_row - 1` in my code.

Reply:
As discussed, I updated the docs.

And luckily I just experienced the panic-inducing case of `max_row = 0` (=> `max_row - 1` panics) and fixed it.